tests/fixtures/sample_lib/modules/sample_task/sample_implementation.py

```diff
@@ -68,19 +70,27 @@ def run(
                 for throw_first_num_requests. Defaults to None.
             throw_first_num_requests (Optional[int], optional): How many requests to throw an error
                 for before being successful. Defaults to None.
+            sleep_time float: How long to sleep before returning a result. Defaults to 0.
+            sleep_increment float: How large of increments to sleep in.
             context (Optional[RuntimeServerContextType]): The context for the runtime server request
         Returns:
             SampleOutputType: The output
         """
         span_name = f"{__name__}.{type(self).__name__}.run"
         with trace.start_child_span(context, span_name):
+            if sleep_time:
```

Comment:
I know @joerunde will not be pleased to introduce additional sleeps in unit tests. Is it possible to get the same behavior using an `Event` instead?